src/patterns/DetailList/DetailList.tsx

```diff
@@ -44,18 +44,22 @@ const DetailList = forwardRef<HTMLDListElement, DetailListProps>(
         {...props}
         ref={ref}
       >
-        {children?.map((child, index) => (
-          <div
-            className={classnames(`${baseClassName}-wrapper`, {
-              [`${px}-has-separators`]: hasSeparators,
-              [`${px}-columns`]: alignment === DetailListAlignment.columns,
-              [`${px}-justified`]: alignment === DetailListAlignment.justified,
-            })}
-            key={getDetailKey(child, index)}
-          >
-            {child}
-          </div>
-        ))}
+        {childrenArray?.map((child) => {
+          const key = isValidElement(child) ? child?.key : null;
```

Comment:
is it guaranteed that child will have a key?

Reply:
I actually think the `key={getDetailKey(child, index)}` is perfect code the right way to do it